DataLogger -> CsvFile

cosparks opened the pull request:
After playing around with DataLogger while building out the calibration display class, I realized that it would make a lot more sense to create a general csv class which provides a means of writing but also READING csv files.

We have to both read from and write to multiple open csv files while running calibration and this seemed like the best way to handle that, so I added:
- ```CsvFile::readRow```: reads a row of a csv file and returns a vector of size CsvFile::_numColumns containing each string entry
- ```CsvFile::readEntry```: reads a single entry from csv file and returns it as a string, incrementing the current entry value of the csv file

I also made some modifications to the class formerly known as DataLogger which will allow us to pass in our own filesystem flags (O_READ, O_RDWR, etc..) so we can more easily control and optimize how we work with the file.

cosparks committed:
Merge branch 'dev-assert' into dev-csv

cosparks committed:
Renames DataLogger name to CsvFile
- changes abstraction: CsvFile represents an open csv file, of which there can be multiple concurrent instances
- adds readEntry and readRow methods to CsvFile
- adds more controllability to CsvFile by allowing user to pass in their own file flags (O_RDWR etc..)
- expands mock filesystem to include implementations of new functionality
- adds unit tests for new CsvFile methods

cosparks committed:
Merge branch 'dev' into dev-csv

cosparks committed:
Revert change to settings.h

cosparks committed:
fixes build error

cosparks committed:
removes recursive calls from CsvFile::create

cosparks commented:
@LaPommeCosmique Sorry, the diff of CsvFile.h/cpp  isn't very useful as it doesn't display new DataLogger changes against old ones.   Let me know if you want me to temporarily change the name back to DataLogger so you can more easier see what I've done there.

cosparks committed:
allow for decimal values and negative entries in CsvFile read methods

LaPommeCosmique commented:
>  temporarily change the name back to DataLogger

Oh ya this would be helpful. I can make that change

LaPommeCosmique committed:
Temporary name change to DataLogger to view diffs

cosparks committed:
Refactoring and Optimizations
- extracts DataLogger::_generateNewFilename and places it in system_util file
- removes _buffer from DataLogger and instead writes all entries directly to _curFile
- modifes logic for adding rows to csv file in DataLogger
- modifies performance tests so that saveToDisk is only called when file is being closed

cosparks committed:
Merge branch 'dev' into dev-csv
